homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -0,0 +1,378 @@
+
+    def __init__(self, name, device, entry, unique_id, coordinator):
+        """Initialize the plug switch."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        if self._model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            self._available_modes = []
+            self._available_modes = [
+                mode.name
+                for mode in AirhumidifierOperationMode
+                if mode is not AirhumidifierOperationMode.Strong
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 10
+        elif self._model in [MODEL_AIRHUMIDIFIER_CA4]:
+            self._available_modes = [
+                mode.name for mode in AirhumidifierMiotOperationMode
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 100
+        else:
+            self._available_modes = [
+                mode.name
+                for mode in AirhumidifierOperationMode
+                if mode is not AirhumidifierOperationMode.Auto
+            ]
+            self._min_humidity = 30
+            self._max_humidity = 80
+            self._humidity_steps = 10
+
+        self._attributes.update({attribute: None for attribute in AVAILABLE_ATTRIBUTES})
+
+    @callback
+    def _handle_coordinator_update(self):
+        """Fetch state from the device."""
+        state = self.coordinator.data
+        if not state:
+            return
+        _LOGGER.debug("Got new state: %s", state)
+        self._available = True
+        self._state = state.is_on
+        self._attributes.update(
+            {
+                key: self._extract_value_from_attribute(state, value)
+                for key, value in AVAILABLE_ATTRIBUTES.items()
+            }
+        )
+        self.async_write_ha_state()
+
+    @property
+    def mode(self):
+        """Return the current mode."""
+        if self._state:
+            return AirhumidifierOperationMode(self._attributes[ATTR_MODE]).name
+        return None
+
+    @property
+    def target_humidity(self):
+        """Return the target humidity."""
+        if self._state:
+            return (
+                self._attributes[ATTR_TARGET_HUMIDITY]
+                if AirhumidifierOperationMode(self._attributes[ATTR_MODE])
+                == AirhumidifierOperationMode.Auto
+                or AirhumidifierOperationMode.Auto.name not in self.available_modes
+                else None
+            )
+        return None
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set the target humidity of the humidifier and set the mode to auto."""
+        target_humidity = self.translate_humidity(humidity)
+        if not target_humidity:
+            return
+
+        _LOGGER.debug("Setting the humidity to: %s", target_humidity)
+        await self._try_command(
+            "Setting operation mode of the miio device failed.",
```

Comment:
Shouldn't this be "Setting the target humidity"?

Reply:
updated